src/group_theory/order_of_element.lean

```diff
@@ -654,10 +654,10 @@ lemma pow_card_eq_one : x ^ fintype.card G = 1 :=
 let ⟨m, hm⟩ := @order_of_dvd_card_univ _ x _ _ in
 by simp [hm, pow_mul, pow_order_of_eq_one]
 
+-- todo: remove `finite_index` hypothesis
 @[to_additive] lemma subgroup.pow_index_mem {G : Type*} [group G] (H : subgroup G)
-  [finite (G ⧸ H)] [normal H] (g : G) : g ^ index H ∈ H :=
-by { casesI nonempty_fintype (G ⧸ H),
-  rw [←eq_one_iff, quotient_group.coe_pow H, index_eq_card, pow_card_eq_one] }
+  [finite_index H] [normal H] (g : G) : g ^ index H ∈ H :=
+by rw [←eq_one_iff, quotient_group.coe_pow H, index_eq_card, pow_card_eq_one]
```

Comment:
This is an easy TODO :)
```suggestion
@[to_additive] lemma subgroup.pow_index_mem {G : Type*} [group G] (H : subgroup G)
  [normal H] (g : G) : g ^ index H ∈ H :=
begin
  by_cases H.index = 0, { rw [h, pow_zero], exact H.one_mem },
  haveI := finite_index.mk h,
  rw [←eq_one_iff, quotient_group.coe_pow H, index_eq_card, pow_card_eq_one],
end
```

Reply:
It is an easy todo, but I was just planning on waiting for #16989, since then you can keep the same 1-line proof (using `pow_card_eq_one'` instead of `pow_card_eq_one`). Would you be happy if I just update to comment to mention #16989?

Reply:
Maybe make this PR depend on #16989?

Reply:
It seems like this refactor more naturally belongs in #16989, so what if I just do the refactor in that PR?

Reply:
https://github.com/leanprover-community/mathlib/pull/16989/commits/7f72d9518367b81d4909cf4714b010a6912da0f4